src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1053,19 +1054,30 @@ RemoveNodeFromCluster(char *nodeName, int32 nodePort)
 	WorkerNode *workerNode = ModifiableWorkerNode(nodeName, nodePort);
 	if (NodeIsPrimary(workerNode))
 	{
+		if (CanRemoveReferenceTablePlacements())
+		{
+			/*
+			 * Delete reference table placements so they are not taken into account
+			 * for the check if there are placements after this.
+			 */
+			DeleteAllReferenceTablePlacementsFromNodeGroup(workerNode->groupId);
+		}
 		bool onlyConsiderActivePlacements = false;
-
-		/*
-		 * Delete reference table placements so they are not taken into account
-		 * for the check if there are placements after this
-		 */
-		DeleteAllReferenceTablePlacementsFromNodeGroup(workerNode->groupId);
-
 		if (NodeGroupHasShardPlacements(workerNode->groupId,
 										onlyConsiderActivePlacements))
 		{
+			if (ClusterHasReferenceTable())
+			{
+				ereport(ERROR, (errmsg(
+									"you cannot remove node because there are reference "
+									"tables and you would lose the data on reference tables"),
+								errhint(
+									"use undistributed_table UDF to undistribute the reference tables")));
+			}
 			ereport(ERROR, (errmsg("you cannot remove the primary node of a node group "
-								   "which has shard placements")));
+								   "which has shard placements"),
+							errhint(
+								"use undistribute_table UDF to undistribute the distributed tables")));
```

Comment:
`use undistribute_table UDF to undistribute the distributed tables` we have a repetition that makes it slightly hard to follow.

So maybe `To proceed, either drop the distributed tables or use undistribute_table() function to convert them to local tables`  and  

`To proceed, either drop the reference tables or use undistribute_table() function to convert them to local tables`